check-mysql/lib/replication.go

```diff
@@ -22,38 +66,65 @@ func checkReplication(args []string) *checkers.Checker {
 	if err != nil {
 		os.Exit(1)
 	}
-	db := newMySQL(opts.mysqlSetting)
-	err = db.Connect()
+	db, err := newDB(opts.mysqlSetting)
 	if err != nil {
-		return checkers.Unknown("couldn't connect DB")
+		return checkers.Unknown(fmt.Sprintf("Couldn't open DB: %s", err))
 	}
 	defer db.Close()
 
-	rows, res, err := db.Query("SHOW SLAVE STATUS")
+	mySQLVersion, err := getMySQLVersion(db)
 	if err != nil {
-		return checkers.Unknown("couldn't execute query")
+		return checkers.Unknown(fmt.Sprintf("Couldn't get MySQL Version: %s", err))
 	}
 
-	if len(rows) == 0 {
-		return checkers.Ok("MySQL is not slave")
+	// MySQL > 8.0.22 supports `SHOW REPLICA STATUS`
+	replicaSupport := !(mySQLVersion.major < 8 || (mySQLVersion.major == 8 && mySQLVersion.minor == 0 && mySQLVersion.patch < 22))
```

Comment:
Does it work properly on MariaDB instances?

Reply:
MariaDB is not considered on this PR.